src/test/java/org/hydev/mcpm/utils/arrays/SlidingWindowTest.java

```diff
@@ -0,0 +1,47 @@
+package org.hydev.mcpm.utils.arrays;
+
+import org.hydev.mcpm.utils.Pair;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SlidingWindowTest {
+
+    ArrayList<Pair<Long, Long>> arr;
+    SlidingWindow slidingWindow;
+    long window;
+
+
+    long naiveSum(long index) {
+        long ret = 0;
+        for (int i = arr.size() - 1; i >= 0; i--) {
+            if (arr.get(i).getKey() < index - window)
+                break;
+            ret += arr.get(i).getValue();
+        }
+        return ret;
+    }
+
+    @Test
+    void testAll() {
+        Random r = new Random();
```

Comment:
We can't have random numbers in a Unit Test like this. This makes the unit tests not deterministic (one call they might succeed another they might now). What you can do is pick a seed and put those numbers in List in a `@BeforeAll` method.

Reply:
well it should always succeed though

Reply:
I do get the idea that you might want the internal behaviour to always be identical, I think this makes the test more "thorough" although harder to debug
If you want I could have it store the seed somewhere so we can see or set it to a fixed seed